### PR TITLE
test: [M3-7069] - Add Cypress integration tests for Linode Create DC-specific pricing

### PR DESCRIPTION
**Pull request by cliu-akamai**
## Description 📝
Add new cypress tests for Linode creation tiered pricing.

## Major Changes 🔄
- Add e2e tests to verify the DC pricing information in Linode creation flow.

## How to test 🧪
```
yarn cy:run -s "cypress/e2e/core/linodes/create-linode.spec.ts"
```

### CODE REVIEW

**Review by abailly-akamai — COMMENTED** [review]
Looks good! couple comments:

- I would consider naming the file `create-linode-dynamic-pricing.spec.ts` so it's clear what the focus of the test is at first sight
- I am getting "The selected region does not have any availability for the requested service plan." when runnign locally. Should we consider getting a much lower plan to run this test to avoid this kind of issue?

**Comment by jdamore-linode** [comment]
> * I would consider naming the file `create-linode-dynamic-pricing.spec.ts` so it's clear what the focus of the test is at first sight.

@abailly-akamai For DC pricing, we're keeping these tests side-by-side with the existing tests because when the feature launches and we clean up the feature flag, all of the new tests will be removed, and the DC-pricing-specific assertions will be refactored into the existing tests.

File: packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts [hunk 1]
```diff
@@ -10,8 +10,32 @@ import { ui } from 'support/ui';
...
+import { makeFeatureFlagData } from 'support/util/feature-flags';
+import {
+  mockGetLinodeType,
+  mockGetLinodeTypes,
+} from 'support/intercepts/linodes';
 
+authenticate();
 describe('create linode', () => {
+  before(() => {
+    cleanUp('linodes');
```
Comment (jdamore-linode) [inline packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts:36]: Great catch 🔍! Thanks @cliu-akamai!

File: packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts [hunk 2]
```diff
@@ -105,4 +129,87 @@ describe('create linode', () => {
...
+    );
+    cy.get('[data-qa-tp="Linode Plan"]').should(
+      'contain.text',
+      dcPricingPlanPlaceholder
+    );
+
+    // Confirm that the checkout summary at the bottom of the page reflects the correct price.
+    containsClick(selectRegionString).type(`${initialRegion.label} {enter}`);
+    fbtClick('Shared CPU');
+    getClick('[id="g6-standard-8"]');
```
Comment (jdamore-linode) [inline packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts:181]:
```suggestion
    getClick(`[id="${dcPricingMockLinodeTypes[0].id}"]`);
```
Coincidentally, I just posted [an explanation on another PR](https://github.com/linode/manager/pull/9707#discussion_r1333483107) about this and why hardcoding the ID here might make the tests fragile in unexpected ways!

File: packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts [hunk 3]
```diff
@@ -105,4 +129,87 @@ describe('create linode', () => {
...
+
+    // TODO: DC Pricing - M3-7086: Uncomment docs link assertion when docs links are added.
+    // cy.findByText(dcPricingDocsLabel)
+    //   .should('be.visible')
+    //   .should('have.attr', 'href', dcPricingDocsUrl);
+
+    // Confirms that the summary updates to reflect price changes if the user changes their region and plan selection.
+    containsClick(initialRegion.label).type(`${newRegion.label} {enter}`);
+    fbtClick('Shared CPU');
+    getClick('[id="g6-standard-8"]');
```
Comment (jdamore-linode) [inline packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts:200]:
```suggestion
    getClick(`[id="${dcPricingMockLinodeTypes[0].id}"]`);
```

**Review by jdamore-linode — APPROVED** [review]
Nice work @cliu-akamai! Everything looks good to me, just a quick suggestion for a couple of your `getClick()` calls!

**Comment by jdamore-linode** [comment]
> * I am getting "The selected region does not have any availability for the requested service plan." when runnign locally. Should we consider getting a much lower plan to run this test to avoid this kind of issue?

@abailly-akamai, are you seeing this message when the test attempts to create the Linode or at another point during the test flow? Haven't been able to reproduce this and can't think of why this might be happening off the top of my head, but I'd like to get to the bottom of it before merging if possible

**Review by bnussman-akamai — APPROVED** [review]
Test passed and looked solid 👍🏼

**Comment by jdamore-linode** [comment]
@cliu-akamai Just circling back here to thank you for those additions! Took a look at those and confirmed they're passing locally.

I think you should be able to fix the automated test failures by merging in the latest changes from `develop`, but if it continues giving us trouble I'll run the suite locally and confirm they're passing

**Comment by abailly-akamai** [comment]
@jdamore-linode sorry for the delay in answering. It happened to me when running the test locally - could it be because my regions list (locally) contains regions the production build CY test against wouldn't, and those regions have no availability (I have a bunch of flags on my account to that effect, ie: `us-east-1b`)?

**Comment by mjac0bs** [comment]
@cliu-akamai I merged in the latest `develop` here and looks like all tests are passing. `linodes/create-linode.spec.ts` also passes locally and with two approvals, I think this is good to merge! 🚢

**Comment by mjac0bs** [comment]
I'm going to merge this now since only one of us seemed to have an issue with local runs, but make a note of this to revisit if it causes more problems with the tests in the future.

**Pull request merged**